pyntc/devices/f5_device.py

```diff
@@ -0,0 +1,480 @@
+
+    def reboot(self, timer=0, confirm=False, volume=None):
+        if confirm:
+            if self._get_active_volume() == volume:
+                volume_name = None
+            else:
+                volume_name = volume
+
+            self._reboot_to_volume(volume_name=volume_name)
+
+            if not self._wait_for_device_reboot(volume_name=volume):
+                raise RuntimeError("Reboot to volume {} failed".format(volume))
+        else:
+            print('Need to confirm reboot with confirm=True')
+
+    def get_boot_options(self):
+        active_volume = self._get_active_volume()
+
+        return {'active_volume': active_volume}
+
+    def set_boot_options(self, image_name, **vendor_specifics):
+        volume = vendor_specifics.get('volume')
+
+        free_space = self._check_free_space(min_space=6)
+
+        if not free_space:
+            raise RuntimeError(
+                "Not enough free space to install OS".format(volume))
+
+        self._image_install(image_name=image_name, volume=volume)
+
+        if not self._wait_for_image_installed(image_name=image_name,
+                                              volume=volume):
+            raise RuntimeError("Installation of {} failed".format(volume))
+
+    def checkpoint(self, filename):
+        raise NotImplementedError
+
+    def rollback(self, checkpoint_file):
+        raise NotImplementedError
+
+    def backup_running_config(self, filename):
+        raise NotImplementedError
+
+    def _get_uptime(self):
+        return self.soap_handler.System.SystemInfo.get_uptime()
+
+    def _get_version(self):
+        return self.soap_handler.System.SystemInfo.get_version()
+
+    def _get_serial_number(self):
+        system_information = self.soap_handler.System.SystemInfo.get_system_information()
+        chassis_serial = system_information.get('chassis_serial')
+
+        return chassis_serial
+
+    def _get_model(self):
+        return self.soap_handler.System.SystemInfo.get_marketing_name()
+
+    def _get_hostname(self):
+        return self.soap_handler.Management.Device.get_hostname(self.devices)[0]
+
+    def _get_interfaces_list(self):
+        interfaces = self.soap_handler.Networking.Interfaces.get_list()
+        return interfaces
+
+    def _get_vlans(self):
+        rd_list = self.soap_handler.Networking.RouteDomainV2.get_list()
+        rd_vlan_list = self.soap_handler.Networking.RouteDomainV2.get_vlan(
+            rd_list)
+
+        return rd_vlan_list
+
+    @staticmethod
+    def _uptime_to_string(uptime):
+        days = uptime / (24 * 60 * 60)
+        uptime = uptime % (24 * 60 * 60)
+        hours = uptime / (60 * 60)
+        uptime = uptime % (60 * 60)
+        mins = uptime / 60
+        uptime = uptime % 60
+        seconds = uptime
+
+        return '%02d:%02d:%02d:%02d' % (days, hours, mins, seconds)
+
+    @property
+    def facts(self):
+        facts = {
+            'uptime': self._get_uptime(),
+            'vendor': 'F5 Networks',
```

Comment:
This should match self.vendor per other device classes

Reply:
Changed this, however vendor assignment is still static as I don't know the method to get it dynamically (is there a real need for it?)